db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -9266,7 +9278,7 @@ else if( key instanceof PermDescriptor)
             }
         }
 
-        dataDictionaryCache.permissionCacheAdd(key, permissions);
+        dataDictionaryCache.permissionCacheAdd(key, permissions==null? Optional.<PermissionsDescriptor>absent():Optional.of(permissions));
```

Comment:
You could use Optional.fromNullable() here (minor nitpick)

Reply:
Thanks @dgomezferro , I will incorporate the suggested change shortly.